src/modules/git_status.rs

```diff
@@ -207,62 +205,64 @@ impl<'a> GitStatusInfo<'a> {
 }
 
 /// Gets the number of files in various git states (staged, modified, deleted, etc...)
-fn get_repo_status(repository: &mut Repository) -> Result<RepoStatus, git2::Error> {
+fn get_repo_status(repo_root: &PathBuf) -> std::io::Result<RepoStatus> {
     log::debug!("New repo status created");
-    let mut status_options = git2::StatusOptions::new();
 
     let mut repo_status = RepoStatus::default();
 
-    match repository.config()?.get_entry("status.showUntrackedFiles") {
-        Ok(entry) => status_options.include_untracked(entry.value() != Some("no")),
-        _ => status_options.include_untracked(true),
-    };
-    status_options
-        .renames_from_rewrites(true)
-        .renames_head_to_index(true)
-        .include_unmodified(true);
-
-    let statuses = repository.statuses(Some(&mut status_options))?;
-
-    if statuses.is_empty() {
-        return Err(git2::Error::from_str("Repo has no status"));
-    }
-
-    statuses
-        .iter()
-        .map(|s| s.status())
-        .for_each(|status| repo_status.add(status));
+    let output = Command::new("git")
+        .current_dir(repo_root)
+        .args(&["config", "--get", "--local", "status.showUntrackedFiles"])
```

Comment:
Just fur my curiosity: Why make this a git config and not a starship one? Means one external call more...

Reply:
The previous code used pulled this same git config option using git2 so I figured I would maintain that for compatibility.  I also thought about that extra external call.  It also makes sense to me for starship to use the git config option, but I can see arguments for both sides.

Reply:
Does `git` not respect its own option here? And if not, it might be faster to read the option directly in starship like before.

Reply:
That is a great point @davidkna.  I would believe git respects its option.  The reason I put this code here was before I made any changes the code was using git2 to pull the same option (so I believe at some point it read it from the repo config).  It may be worth just removing this altogether.

The starship configuration for showing the untracked files in the prompt applies independently.  The previous code pulled in all the information and then selectively pulled that information to put into the prompt, which is the same thing I am doing here.

Actually, now that I think about it, the way I am getting the current status removes the need for this command.  I think the old code had to supply this because it used git2 which may not have respected the current configuration.  I will remove this.